api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -1,173 +1,60 @@
-  | Debug
-  | Debug2
-  | Debug3
-  | Debug4
-  | Info
-  | Info2
-  | Info3
-  | Info4
-  | Warn
-  | Warn2
-  | Warn3
-  | Warn4
-  | Error
-  | Error2
-  | Error3
-  | Error4
-  | Fatal
-  | Fatal2
-  | Fatal3
-  | Fatal4
-  | Unknown !Int32
-  deriving (Eq, Ord, Read, Show )
+{- | Emits a LogRecord with properties specified by the passed in Logger and LogRecordArguments.
+If observedTimestamp is not set in LogRecordArguments, it will default to the current timestamp.
+If context is not specified in LogRecordArguments it will default to the current context.
+-}
+emitLogRecord
+  :: (MonadIO m)
+  => Logger
+  -> LogRecordArguments body
+  -> m (LogRecord body)
+emitLogRecord logger LogRecordArguments {..} = do
+  currentTimestamp <- getCurrentTimestamp
+  let logRecordObservedTimestamp = fromMaybe currentTimestamp observedTimestamp
 
-instance Enum SeverityNumber where
-  toEnum 1 = Trace
-  toEnum 2 = Trace2
-  toEnum 3 = Trace3
-  toEnum 4 = Trace4
-  ... -}
+  logRecordTracingDetails <- runMaybeT $ do
+    currentContext <- liftIO getContext
+    currentSpan <- MaybeT $ pure $ lookupSpan $ fromMaybe currentContext context
+    SpanContext {traceId, spanId, traceFlags} <- getSpanContext currentSpan
+    pure (traceId, spanId, traceFlags)
 
--- severityTrace :: SeverityNumber
--- severityTrace = 1
--- severityTrace2 :: SeverityNumber
--- severityTrace2 = 2
--- severityTrace3 :: SeverityNumber
--- severityTrace3 = 3
--- severityTrace4 :: SeverityNumber
--- severityTrace4 = 4
--- severityDebug :: SeverityNumber
--- severityDebug = 5
--- severityDebug2 :: SeverityNumber
--- severityDebug2 = 6
--- severityDebug3 :: SeverityNumber
--- severityDebug3 = 7
--- severityDebug4 :: SeverityNumber
--- severityDebug4 = 8
--- severityInfo :: SeverityNumber
--- severityInfo = 9
--- severityInfo2 :: SeverityNumber
--- severityInfo2 = 10
--- severityInfo3 :: SeverityNumber
--- severityInfo3 = 11
--- severityInfo4 :: SeverityNumber
--- severityInfo4 = 12
--- severityWarn :: SeverityNumber
--- severityWarn = 13
--- severityWarn2 :: SeverityNumber
--- severityWarn2 = 14
--- severityWarn3 :: SeverityNumber
--- severityWarn3 = 15
--- severityWarn4 :: SeverityNumber
--- severityWarn4 = 16
--- severityError :: SeverityNumber
--- severityError = 17
--- severityError2 :: SeverityNumber
--- severityError2 = 18
--- severityError3 :: SeverityNumber
--- severityError3 = 19
--- severityError4 :: SeverityNumber
--- severityError4 = 20
--- severityFatal :: SeverityNumber
--- severityFatal = 21
--- severityFatal2 :: SeverityNumber
--- severityFatal2 = 22
--- severityFatal3 :: SeverityNumber
--- severityFatal3 = 23
--- severityFatal4 :: SeverityNumber
--- severityFatal4 = 24
+  pure
+    LogRecord
+      { logRecordTimestamp = timestamp
+      , logRecordObservedTimestamp
+      , logRecordTracingDetails
+      , logRecordSeverityNumber = fmap mkSeverityNumber severityNumber
+      , logRecordSeverityText = severityText <|> (shortName . mkSeverityNumber =<< severityNumber)
```

Comment:
Uses the supplied `severityText` if there is some, but uses the `shortName` from the supplied `severityNumber` if there is not.

Reply:
Should it be the other way around? I'm not clear on which one is actually higher precedence here.

Reply:
In the docs for `SeverityText`:
> If this field is missing and `SeverityNumber` is present then the short name that corresponds to the `SeverityNumber` may be used as a substitution.

https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitytext